src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs

```diff
@@ -0,0 +1,202 @@
+                    matchedActionBuilder.Update(new Dictionary<string, Column>()));
+                Assert.IsType<DeltaMergeBuilder>(
+                    matchedActionBuilder.UpdateExpr(new Dictionary<string, string>()));
+                Assert.IsType<DeltaMergeBuilder>(matchedActionBuilder.UpdateAll());
+                Assert.IsType<DeltaMergeBuilder>(matchedActionBuilder.Delete());
+
+                // Validate the MergeBuilder not-matched signatures.
+                Assert.IsType<DeltaMergeNotMatchedActionBuilder>(mergeBuilder.WhenNotMatched());
+                Assert.IsType<DeltaMergeNotMatchedActionBuilder>(
+                    mergeBuilder.WhenNotMatched("id = 5"));
+                DeltaMergeNotMatchedActionBuilder notMatchedActionBuilder =
+                    Assert.IsType<DeltaMergeNotMatchedActionBuilder>(
+                        mergeBuilder.WhenNotMatched(Functions.Expr("id = 5")));
+
+                Assert.IsType<DeltaMergeBuilder>(
+                    notMatchedActionBuilder.Insert(new Dictionary<string, Column>()));
+                Assert.IsType<DeltaMergeBuilder>(
+                    notMatchedActionBuilder.InsertExpr(new Dictionary<string, string>()));
+                Assert.IsType<DeltaMergeBuilder>(notMatchedActionBuilder.InsertAll());
+
+                // Update and UpdateExpr should return void.
+                table.Update(new Dictionary<string, Column>() { });
+                table.Update(Functions.Expr("id % 2 == 0"), new Dictionary<string, Column>() { });
+                table.UpdateExpr(new Dictionary<string, string>() { });
+                table.UpdateExpr("id % 2 == 1", new Dictionary<string, string>() { });
+
+                Assert.IsType<DataFrame>(table.Vacuum());
+                Assert.IsType<DataFrame>(table.Vacuum(168));
+
+                // Delete should return void.
+                table.Delete("id > 10");
+                table.Delete(Functions.Expr("id > 5"));
+                table.Delete();
+            }
+        }
+
+        /// <summary>
+        /// Validate that a tutorial DataFrame contains only the expected values.
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <param name="dataFrame"></param>
+        private void ValidateTutorialDataFrame(
+            IEnumerable<int> expectedValues,
+            DataFrame dataFrame)
+        {
+            Assert.Equal(expectedValues.Count(), dataFrame.Count());
+
+            var sortedExpectedValues = new List<int>(expectedValues);
+            sortedExpectedValues.Sort();
+
+            var sortedValues = new List<int>(
+                dataFrame
+                    // We need to select the "id" column, otherwise Collect() won't show the
+                    // updates made to the DeltaTable.
+                    .Select("id")
+                    .Sort("id")
+                    .Collect()
+                    .Select(r => Convert.ToInt32(r.Get("id"))));
```

Comment:
I would do `row` instead of `r` here to be explicit.

Reply:
Changed.